### PR TITLE
Add HTTP-CORS ballerina by example

### LINKED ISSUE #3975: Document CORS support
There is no documentation on CORS support in ballerina. Can we document this? Maybe we can add a sample.

### PR DESCRIPTION
**Pull request by chamil321**
PR adds ballerina by example to depict how to use CORS in a ballerina server connector.
Fixes #3975

### CODE REVIEW

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java [hunk 1]
```diff
@@ -108,22 +108,26 @@ public static void process(HTTPCarbonMessage requestMsg, HTTPCarbonMessage respo
...
+                    "multiple request methods";
+            bLog.info(action + error);
             return null;
         }
         String requestMethod = requestMethods.get(0);
         CorsHeaders resourceCors = getResourceCors(cMsg, requestMethod);
         if (resourceCors == null || !resourceCors.isAvailable()) {
-            bLog.info(action + "headers are not declared properly");
+            String error = resourceCors == null ? "method not allowed" :
+                    "cors headers not declared properly";
```
Comment (shafreenAnfar) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java]: cors or CORS ?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java [hunk 2]
```diff
@@ -108,22 +108,26 @@ public static void process(HTTPCarbonMessage requestMsg, HTTPCarbonMessage respo
         //6.2.3 - request must have access-control-request-method, must be single-valued
         List<String> requestMethods = getHeaderValues(Constants.AC_REQUEST_METHOD, cMsg);
         if (requestMethods == null || requestMethods.size() != 1) {
-            bLog.info(action + "not allowed request methods");
+            String error = requestMethods == null ? "Access-Control-Request-Method header is unavailable" :
+                    "multiple request methods";
```
Comment (shafreenAnfar) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java]: "multiple request methods" is it informative enough ?

File: samples/ballerina-by-example/examples/http-cors/http-cors.bal [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import ballerina.net.http;
+
+@Description {value:"Service level CORS headers applies globally"}
```
Comment (shafreenAnfar) [inline samples/ballerina-by-example/examples/http-cors/http-cors.bal]: Shall we say "Service level CORS headers applies globally for each resource" ?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java (hunk 1 above)
Comment (chamil321) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java]: Changed to CORS

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java (hunk 2 above)
Comment (chamil321) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/CorsHeaderGenerator.java]: Changed

File: samples/ballerina-by-example/examples/http-cors/http-cors.bal (hunk 3 above)
Comment (chamil321) [reply samples/ballerina-by-example/examples/http-cors/http-cors.bal]: Ack

**Pull request merged**